pyfr/writers/nodemaps.py

```diff
@@ -0,0 +1,386 @@
+# -*- coding: utf-8 -*-
+
+"""Mappings between the ordering of PyFR nodes, and those of external formats
+
+"""
+import numpy as np
+
+
+class VTK8NodeMaps(object):
+    """Mappings between the node ordering of PyFR and that of VTK
+    Lagrange cells
+
+    Node mappings are contained within two dictionaries; one maps from
+    VTK node ordering to PyFR, and the other provides the inverse.
+    Dictionary items are keyed by a tuple of element type (string) and
+    number of solution points per element (integer).
+
+    INFO: VTK updated the numbering of Lagrange hex elements in VTK9
+    due to a bug in a documentation (but not a bug in the implementation), see
+    https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6678
+    If the *.vtu or *.pvtu files are written using xml version < 5.1,
+    then VTK and Paraview will automatically suppose that the node numbering
+    is consistent with the one found in VTK8 and will automatically renumber
+    Lagrange hex elements without the users being aware of it.
+    PyFR hardcodes the token xml version to 1.0, therefore we need to use
+    the VTK8 connectivity.
+    The mappings between PyFR to VTK9 are also added for the
+    sake of completeness.
+    """
+
+    from_pyfr = {
+        ('quad', 4): np.array([0, 1, 3, 2]),
+        ('quad', 9): np.array([0, 2, 8, 6, 1, 5,
+                               7, 3, 4]),
+        ('quad', 16): np.array([0,  3, 15, 12,  1,
+                                2,  7, 11, 13, 14,
+                                4,  8,  5,  6,  9,
+                                10]),
+        ('quad', 25): np.array([0,  4, 24, 20,  1,
+                                2,  3,  9, 14, 19,
+                                21, 22, 23,  5, 10,
+                                15,  6,  7,  8, 11,
+                                12, 13, 16, 17, 18]),
+        ('quad', 36): np.array([0,  5, 35, 30,  1,
+                                2,  3,  4, 11, 17,
+                                23, 29, 31, 32, 33,
+                                34,  6, 12, 18, 24,
+                                7,  8,  9, 10, 13,
+                                14, 15, 16, 19, 20,
+                                21, 22, 25, 26, 27,
+                                28]),
+        ('hex', 8): np.array([0, 1, 3, 2, 4, 5,
```

Comment:
Can be a bit more aggressive with the line lengths.

Reply:
Ok I'll try to reduce the line length. Do you think that this code should go to `readers.nodemaps` instead?

Reply:
I think the line can be made longer and still be within 79 characters.  I do not have a strong preference as to where it goes; someday we might tabulate these things in files like we do with quadrature rules.